core/src/services/sftp/writer.rs

```diff
@@ -15,38 +15,51 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use std::pin::Pin;
+use std::task::Context;
+use std::task::Poll;
+
 use async_trait::async_trait;
 use openssh_sftp_client::file::File;
+use openssh_sftp_client::file::TokioCompatFile;
+use tokio::io::AsyncWrite;
 
 use crate::raw::oio;
 use crate::*;
 
 pub struct SftpWriter {
-    file: File,
+    file: Pin<Box<TokioCompatFile>>,
 }
 
 impl SftpWriter {
     pub fn new(file: File) -> Self {
-        SftpWriter { file }
+        SftpWriter {
+            file: Box::pin(TokioCompatFile::new(file)),
+        }
     }
 }
 
 #[async_trait]
 impl oio::Write for SftpWriter {
-    async fn write(&mut self, bs: &dyn oio::WriteBuf) -> Result<usize> {
-        let size = self.file.write(bs.chunk()).await?;
+    fn poll_write(&mut self, cx: &mut Context<'_>, bs: &dyn oio::WriteBuf) -> Poll<Result<usize>> {
+        self.file
+            .as_mut()
+            .poll_write(cx, bs.chunk())
+            .map_err(parse_io_error)
+    }
 
-        Ok(size)
+    fn poll_close(&mut self, cx: &mut Context<'_>) -> Poll<Result<()>> {
+        self.file.as_mut().poll_shutdown(cx).map_err(parse_io_error)
     }
 
-    async fn abort(&mut self) -> Result<()> {
-        Err(Error::new(
+    fn poll_abort(&mut self, _: &mut Context<'_>) -> Poll<Result<()>> {
+        Poll::Ready(Err(Error::new(
             ErrorKind::Unsupported,
-            "SFTP does not support aborting writes",
-        ))
+            "HdfsWriter doesn't support abort",
```

Comment:
This should be "SftpWriter doesn't support abort"

Reply:
> This should be "SftpWriter doesn't support abort"

Oh, my bad. Would you like to help fix this typo?